bson/_cbsonmodule.c

```diff
@@ -1854,8 +1923,50 @@ static PyObject* get_value(PyObject* self, PyObject* name, const char* buffer,
             }
             memcpy(&millis, buffer + *position, 8);
             millis = (int64_t)BSON_UINT64_FROM_LE(millis);
-            naive = datetime_from_millis(millis);
             *position += 8;
+
+            if (options->datetime_conversion == DATETIME_CONVERSION_OPTS_DATETIME_MS){
+                value = datetime_ms_from_millis(self, millis);
+                break;
+            }
+
+            int dt_clamp = options->datetime_conversion == DATETIME_CONVERSION_OPTS_DATETIME_CLAMP;
+            int dt_auto = options->datetime_conversion == DATETIME_CONVERSION_OPTS_DATETIME_AUTO;
+
+
+            if (dt_clamp || dt_auto){
+                PyObject *min_millis_fn = _get_object(state->_min_datetime_ms, "bson.datetime_ms", "_min_datetime_ms");
```

Comment:
All `_get_object` calls need the return value to be NULL checked for error handling.